### ISSUE TITLE
Api<Unit, Throwable> crashes with NPE when http code is 204 or 205

### ISSUE THREAD

**Issue by sky-josemessejana** [slackhq/EitherNet#62]
### Describe the bug
When using Api<Unit, Throwable> and the response is a http 204 or 2005 the app crashes with a NPE with the following stacktrace
```
at retrofit2.KotlinExtensions$await$2$2.onResponse(KotlinExtensions.kt:43)
at com.slack.eithernet.ApiResultCallAdapterFactory$ApiResultCallAdapter$adapt$1$enqueue$1.onResponse(ApiResult.kt:313)
at retrofit2.OkHttpCall$1.onResponse(OkHttpCall.java:161)
at okhttp3.internal.connection.RealCall$AsyncCall.run(RealCall.kt:519)
```

This is due to retrofit passing a null body on OkHttpCall.java #parseResponse when http code is 204 or 205. ApiResult will then call                   callback.onResponse(call, Response.success(withTag)) with null `withTag`, which will then crash on retrofit's KotlinExtensions

### Requirements (place an `x` in each of the `[ ]`)**
* [X] I've read and understood the [Contributing guidelines](../CONTRIBUTING.md) and have done my best effort to follow them.
* [X] I've read and agree to the [Code of Conduct](https://slackhq.github.io/code-of-conduct).
* [X] I've searched for any related issues and avoided creating a duplicate issue.

### To Reproduce
Have a retrofit method that returns ApiResult<Unit, Throwable> and the response returns a http code 204 or 205.

### Expected behavior
App behaves normally as it does with http error 200s

#### Reproducible in: 1.4.0

Project version:

OS version(s): 12

**Comment by ZacSweers** [comment]
Thanks for the report. Do you want to take a shot at a PR to fix? Seems like it could be easy enough to check for null in those error codes

**Comment by JDSM01** [comment]
Hi, open PR for the fix https://github.com/slackhq/EitherNet/pull/63. Can you please review 🙏

**Comment by ZacSweers** [comment]
Fixed in #63
